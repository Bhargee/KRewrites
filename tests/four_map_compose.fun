map (fun a -> a) (map (fun b -> b)) (map (fun c -> c)) (map (fun d -> d / 5))
