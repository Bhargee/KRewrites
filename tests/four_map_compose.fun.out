Search results:

Solution 1:
<T>
    <k>
        ((map) ((((fun a -> a) -| (fun b -> b)) -| (fun c -> c)) -| (fun d
           -> (d / 5))))
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 2:
<T>
    <k>
        ((map) (((fun a -> a) -| ((fun b -> b) -| (fun c -> c))) -| (fun d
           -> (d / 5))))
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 3:
<T>
    <k>
        ((map) (((fun a -> a) -| (fun b -> b)) -| ((fun c -> c) -| (fun d 
          -> (d / 5)))))
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>
